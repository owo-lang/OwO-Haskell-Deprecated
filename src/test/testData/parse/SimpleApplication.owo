module SimpleApplication where
 a = f a b
 a = f (a b)
 a = (f a) b
 undefined = undefined