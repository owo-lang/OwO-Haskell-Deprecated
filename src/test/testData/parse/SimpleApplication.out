File type: owo
Top module name: .SimpleApplication
 implementation of function  a (32 2 2) (33 2 3)
  no pragmas
  clause for  a (32 2 2) (33 2 3)
  patterns matching parsed as expression
   named reference a (32 2 2) (33 2 3)
  function body
   function application
   function being applied
    function application
    function being applied
     named reference f (36 2 6) (37 2 7)
    value applied to the function
     named reference a (38 2 8) (39 2 9)
   value applied to the function
    named reference b (40 2 10) (41 2 11)
  no with abstraction
  no where clause
  clause for  a (43 3 2) (44 3 3)
  patterns matching parsed as expression
   named reference a (43 3 2) (44 3 3)
  function body
   function application
   function being applied
    named reference f (47 3 6) (48 3 7)
   value applied to the function
    function application
    function being applied
     named reference a (50 3 9) (51 3 10)
    value applied to the function
     named reference b (52 3 11) (53 3 12)
  no with abstraction
  no where clause
  clause for  a (56 4 2) (57 4 3)
  patterns matching parsed as expression
   named reference a (56 4 2) (57 4 3)
  function body
   function application
   function being applied
    function application
    function being applied
     named reference f (61 4 7) (62 4 8)
    value applied to the function
     named reference a (63 4 9) (64 4 10)
   value applied to the function
    named reference b (66 4 12) (67 4 13)
  no with abstraction
  no where clause
 implementation of function  undefined (69 5 2) (78 5 11)
  no pragmas
  clause for  undefined (69 5 2) (78 5 11)
  patterns matching parsed as expression
   named reference undefined (69 5 2) (78 5 11)
  function body
   named reference undefined (81 5 14) (90 5 23)
  no with abstraction
  no where clause
